### DISCUSSION TITLE
Storage: public paths

### DISCUSSION THREAD

**Discussion by joshnuss** [supabase/supabase#1003] [Questions] [answered]
The result from querying storage returns a list of objects that contains `name` etc.. but it doesn't include a `path`

```javascript
await supabase.storage
  .from('my-bucket-name')
  .list('products/images', {limit: 20})
```

```json5
[
  {
    "name": "t-shirt.png",
    "id": "abcd-123",
    "updated_at": "2021-03-31T02:33:39.616739+00:00",
    "created_at": "2021-03-31T02:33:39.616739+00:00",
    "last_accessed_at": "2021-03-31T02:33:39.616739+00:00",
    "metadata": {
      "size": 1000,
      "mimetype": "image/png",
      "cacheControl": "no-cache"
    }
  },
  // ...
 ]
```

If the `object`'s policy is set to public read, it would be handy to have a stable URL to link to any images, videos, pdf to.

Maybe it can be accomplished with a proxy that caches the file on first-request if it's public? (according to cache policy)

**Comment by inian** [comment, marked as answer]
Hi @joshnuss, yup currently the API doesn't prefix the folder when returning the results. So in this case the full path for the object would be `product/images/t-shirt.png`

For creating stable URLs, are you looking for [signed URLs](https://supabase.io/docs/reference/javascript/storage-from-createsignedurl)? If you set a long enough expiry, you can freely link or embed the object.

**Reply by joshnuss** [reply]
Thanks @inian, signed URLs with a long expiration would work. cheers!
